airflow/api_connexion/endpoints/request_dict.py

```diff
@@ -15,12 +15,13 @@
 # specific language governing permissions and limitations
 # under the License.
 from __future__ import annotations
-
+from marshmallow import ValidationError
 from typing import Any, Mapping, cast
 
 
 def get_json_request_dict() -> Mapping[str, Any]:
     """Cast request dictionary to JSON."""
     from flask import request
-
+    if request.data == b'':
+        raise ValidationError("POST Body must not be None")
```

Comment:
```python
    if not request.data:
```

Does this work?

Also, why raise a ValidationError that is simply always caught, instead of raising BadRequest directly?

Reply:
A) Yes this works, will change - much better than using hard-coded compare.
B) I was under the impression that BadRequest shouldn't be used inside internal services and only in the handlers, but makes sense to use it because it's in the api_connexion folder.
Will do all these changes, one last question the static run is failing, I found the error but do I need to regenerate generated ts file too?

Reply:
Changes done, hopefully, pre-commit should pass now.

Reply:
+ Please ignore the spam commits, I was testing something and it went wrong due to which it added a bunch of reviewers.